catalog/service/impl/src/main/java/org/projectnessie/catalog/service/impl/SignerKeysServiceImpl.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.catalog.service.impl;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+import static java.util.UUID.randomUUID;
+
+import com.google.common.annotations.VisibleForTesting;
+import jakarta.enterprise.context.RequestScoped;
+import jakarta.inject.Inject;
+import java.security.NoSuchAlgorithmException;
+import java.security.SecureRandom;
+import java.time.Clock;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Base64;
+import java.util.List;
+import java.util.concurrent.ThreadLocalRandom;
+import org.projectnessie.catalog.service.api.SignerKeysService;
+import org.projectnessie.catalog.service.objtypes.ImmutableSignerKeysObj;
+import org.projectnessie.catalog.service.objtypes.SignerKey;
+import org.projectnessie.catalog.service.objtypes.SignerKeysObj;
+import org.projectnessie.versioned.storage.common.exceptions.ObjNotFoundException;
+import org.projectnessie.versioned.storage.common.exceptions.ObjTooLargeException;
+import org.projectnessie.versioned.storage.common.objtypes.UpdateableObj;
+import org.projectnessie.versioned.storage.common.persist.Persist;
+
+/**
+ * Signer keys service implementation based on {@link Persist} using a {@link SignerKeysObj} per
+ * Nessie repository. Relies on Nessie's distributed cache invalidation, when used in a horizontally
+ * scaled setup (multiple Nessie processes serving the same repository).
+ */
+@RequestScoped
+public class SignerKeysServiceImpl implements SignerKeysService {
+  public static final long SPIN_LOOP_MIN_SLEEP_MILLIS = 20;
+  public static final long SPIN_LOOP_MAX_SLEEP_MILLIS = 200;
+  public static final Duration NEW_KEY_ROTATE_AFTER = Duration.of(3, DAYS);
```

Comment:
Should this be configurable?

Reply:
Yea - want to do that as a follow-up after #9170